File: 1550_Three_Consecutive_Odds.py
class Solution:
    def threeConsecutiveOdds(self, arr: list[int]) -> bool:
        count = 0

        for num in arr:
            if num % 2 != 0:
                count += 1
            else:
                count = 0

            if count == 3:
                return True

        return False
        

a = Solution()
print(a.threeConsecutiveOdds([2, 6, 4, 1]))  # False
print(a.threeConsecutiveOdds([1, 2, 3, 4]))  # False
print(a.threeConsecutiveOdds([1, 2, 3, 5, 7]))  # True

File: 1550_Three_Consecutive_Odds.c
#include <stdbool.h>
#include <stdio.h>
bool threeConsecutiveOdds(int* arr, int arrSize) {
    int count = 0;
    for(int i = 0; i < arrSize; i++){
        if(arr[i] % 2 != 0){
            count++;
        }
        else{
            count = false;
        }

        if(count == 3){
            return true;
        }
    }

    return 0;
}

int main() {
    int arr[] = {1, 2, 3, 4, 5, 7, 9};
    int arrSize = sizeof(arr) / sizeof(arr[0]);
    bool result = threeConsecutiveOdds(arr, arrSize);
    printf("%s\n", result ? "true" : "false");
    return 0;
}